### PR TITLE
test(local): overhaul test suite

### PR DESCRIPTION
**Pull request by jbrockopp**
Part of the effort for https://github.com/go-vela/community/issues/54

This refactors the tests we have setup for the `go-vela/pkg-executor/executor/local` package.

The test coverage went from `87.4%` to `89.9%` and the time to execute went from ~ `16s` to ~ `6s`

You'll also note that the total number of LOC for the tests was cut in half.

These are the results of test coverage on the `master` branch:

```sh
 $ make test-cover

### Creating test coverage report
?   	github.com/go-vela/pkg-executor/cmd/vela-executor	[no test files]
ok  	github.com/go-vela/pkg-executor/executor	0.570s	coverage: 100.0% of statements
ok  	github.com/go-vela/pkg-executor/executor/linux	18.602s	coverage: 83.4% of statements
ok  	github.com/go-vela/pkg-executor/executor/local	15.998s	coverage: 87.4% of statements
?   	github.com/go-vela/pkg-executor/internal	[no test files]
ok  	github.com/go-vela/pkg-executor/internal/build	0.243s	coverage: 100.0% of statements
ok  	github.com/go-vela/pkg-executor/internal/service	0.360s	coverage: 100.0% of statements
ok  	github.com/go-vela/pkg-executor/internal/step	0.479s	coverage: 100.0% of statements

### Opening test coverage report
```

These are the results of test coverage on the `test/local/overhaul` branch:

```sh
$ make test-cover

### Creating test coverage report
?   	github.com/go-vela/pkg-executor/cmd/vela-executor	[no test files]
ok  	github.com/go-vela/pkg-executor/executor	0.550s	coverage: 100.0% of statements
ok  	github.com/go-vela/pkg-executor/executor/linux	16.053s	coverage: 83.4% of statements
ok  	github.com/go-vela/pkg-executor/executor/local	6.151s	coverage: 89.9% of statements
?   	github.com/go-vela/pkg-executor/internal	[no test files]
ok  	github.com/go-vela/pkg-executor/internal/build	0.618s	coverage: 100.0% of statements
ok  	github.com/go-vela/pkg-executor/internal/service	0.782s	coverage: 100.0% of statements
ok  	github.com/go-vela/pkg-executor/internal/step	0.392s	coverage: 100.0% of statements

### Opening test coverage report
```

### CODE REVIEW

File: executor/local/stage_test.go [hunk 1]
```diff
@@ -44,58 +44,40 @@
~	}
~
 	// setup tests
 	tests := []struct {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/local/stage_test.go:47]: Ignoring this 👍

File: executor/local/service_test.go [hunk 2]
```diff
@@ -216,35 +188,42 @@
...
 			},
 		},
+		{ // empty service container
+			failure:   true,
+			container: new(pipeline.Container),
+		},
 	}
 
 	// run tests
 	for _, test := range tests {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/local/service_test.go:212]: Ignoring this 👍

File: executor/local/step_test.go [hunk 3]
```diff
@@ -212,77 +180,76 @@
...
 			},
 		},
+		{ // empty step container
+			failure:   true,
+			container: new(pipeline.Container),
+		},
 	}
 
 	// run tests
 	for _, test := range tests {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/local/step_test.go:239]: Ignoring this 👍

File: executor/local/build_test.go [hunk 4]
```diff
@@ -141,30 +117,33 @@
~	}
 
 	// run test
 	for _, test := range tests {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/local/build_test.go:120]: Ignoring this 👍

File: executor/local/build_test.go [hunk 5]
```diff
@@ -209,65 +178,72 @@
...
-			pipeline: "testdata/build/services/img_notfound.yml",
-		},
-		{ // pipeline with service image tag ignoring not found
-			failure:  true,
-			pipeline: "testdata/build/services/img_ignorenotfound.yml",
-		},
 	}
 
 	// run test
 	for _, test := range tests {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/local/build_test.go:220]: Ignoring this 👍

File: executor/local/build_test.go [hunk 6]
```diff
@@ -472,61 +366,60 @@
...
-		{ // pipeline with service image tag not found
+		{ // stages pipeline with name not found
 			failure:  false,
-			pipeline: "testdata/build/services/img_notfound.yml",
+			pipeline: "testdata/build/stages/name_notfound.yml",
 		},
 	}
 
 	// run test
 	for _, test := range tests {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/local/build_test.go:396]: Ignoring this 👍

**Review by matt-fevold — APPROVED** [review]
increased coverage, lower LOC AND faster runtime? you sir are a snake oil salesman.
LGTM 🐕

**Review by kneal — APPROVED** [review]
LGTM 🐬

**Pull request merged**